pandas/tests/groupby/test_categorical.py

```diff
@@ -506,6 +506,23 @@ def test_observed_groups(observed):
     tm.assert_dict_equal(result, expected)
 
 
+def test_groups_na_category(dropna, observed):
+    # https://github.com/pandas-dev/pandas/issues/61356
+    df = DataFrame(
+        {"cat": Categorical(["a", np.nan, "a"], categories=list("adb"))},
+        index=list("xyz"),
+    )
+    g = df.groupby("cat", observed=observed, dropna=dropna)
+
+    result = g.groups
+    expected = {"a": Index(["x", "z"])}
+    if not dropna:
+        expected |= {np.nan: Index(["y"])}
```

Comment:
When both arguments are False, should NaN come after non-observed groups? That seems more intuitive to me, especially for an ordered categorical

Reply:
No - if you do an operation like sum the order here matches the order in that result.

Reply:
This is what I'm getting on both main and 2.2.3.

```
>>> df = DataFrame(
...         {"cat": Categorical(["a", np.nan, "a"], categories=list("adb"))},
...         index=list("xyz"),
...     )
>>> df["val"] = [1, 2, 3]
>>> g = df.groupby("cat", observed=False, dropna=False)
>>> g.sum()
     val
cat
a      4
d      0
b      0
NaN    2
```

Reply:
Ah, `tm.assert_dict_equal` appears to be order-invariant, so it doesn't matter for the test.

Reply:
Ah, I see now. I was correct in that the order was the same, but I failed to notice that the test added the columns in the incorrect order. I do wonder if `assert_dict_equal` should default to checking the order (perhaps with an argument to ignore order).